### PR TITLE
Add some more content-based matter tests.

### PR DESCRIPTION
**Pull request by tecimovic**

### CODE REVIEW

**Comment by brdandu** [comment]
Other templates to consider https://github.com/project-chip/connectedhomeip/tree/master/src/app/zap-templates/templates/app

File: test/gen-matter-4.test.js [hunk 1]
```diff
@@ -83,6 +83,80 @@ test(
       { disableDeprecationWarnings: true }
     )
     expect(genResult.hasErrors).toEqual(false)
+
+    // Note: A lot of these tests here are for sake of backwards
+    // compatibility. Latest ZAP must be able to generate content for
+    // the old SDKs, so if you changed something that generates
+    // endpoint_config differently, please be very very careful and
+    // make yourself you can answer positively the following question:
```
Comment (bzbarsky-apple) [inline test/gen-matter-4.test.js:91]:
```suggestion
    // make sure you can answer positively the following question:
```

File: test/gen-matter-4.test.js [hunk 2]
```diff
@@ -83,6 +83,80 @@ test(
...
+    //   will after my changes, zap still be able to generate content
+    //   that will work with an older SDK.
+    //
+    let ept = genResult.content['endpoint_config.h']
+    expect(ept).toContain(`{ \\
+      /* Endpoint: 65534, Cluster: Network Commissioning (server) */ \\
```
Comment (bzbarsky-apple) [inline test/gen-matter-4.test.js:97]: Shouldn't we also have some tests for non-default `order` values?  Here and in other places.

File: test/gen-matter-4.test.js (hunk 1 above)
Comment (tecimovic) [reply test/gen-matter-4.test.js:91]: Right....

File: test/gen-matter-4.test.js (hunk 2 above)
Comment (tecimovic) [reply test/gen-matter-4.test.js:97]: I'll add some tests for that `order="...."` into the template.

**Pull request merged**